### PR TITLE
[receiver/windowseventlog] Fix buffer overflow when reading raw events

### PR DESCRIPTION
**Pull request by swiatekm**
**Description:**
Fixing a bug where an event larger than half the buffer capacity would cause a panic due to an overflow. The reason this happens is because raw events are render using the `evtRender` syscall, which returns the byte count, rather than the UTF16 character count, but we treat it as if it was the latter.

**Link to tracking Issue:** #23677

**Testing:** 
Added some tests for previously untested parts of the code, and a test that triggers the problem in particular. I've also done a E2E test on a live system which previously triggered the problem, but doesn't after the fix.

### CODE REVIEW

File: pkg/stanza/operator/input/windows/event_test.go [hunk 1]
```diff
@@ -7,7 +7,12 @@
 package windows
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
+	"unsafe"
+
+	"golang.org/x/text/encoding/unicode"
```
Comment (djaglowski) [inline pkg/stanza/operator/input/windows/event_test.go:15]: I think this will fail lint because it should be in one group with the testify package.

**Review by djaglowski — COMMENTED** [review]
Thanks for fixing this @swiatekm-sumo.

I've added the `Run Windows` tag to the PR to trigger CI on windows. Looks like there are a couple failures, but I think the approach looks correct.

**Comment by swiatekm-sumo** [comment]
> Thanks for fixing this @swiatekm-sumo.
> 
> I've added the `Run Windows` tag to the PR to trigger CI on windows. Looks like there are a couple failures, but I think the approach looks correct.

Yeah, looks like checkptr doesn't like my beautiful pointer arithmetic in the mock I added for tests. I'll try to massage it into compliance.

**Comment by swiatekm-sumo** [comment]
I ended up dropping the tests for now. Mocking the Windows syscalls the way I wanted broke the rules which the Go compiler enforces on unsafe pointer usage. I'll need to change the `Event` struct to use dependency injection for the syscall api, and I don't want to hold this fix up with that.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**